mmaction/datasets/pipelines/formatting.py

```diff
@@ -374,6 +374,64 @@ def __repr__(self):
         return repr_str
 
 
+@PIPELINES.register_module()
+class JointToBone:
+    """Convert the joint information to bone information.
+
+    Required keys are "keypoint" ,
+    added or modified keys are "keypoint".
+
+    Args:
+        dataset (str): Define the type of dataset.
+    """
+
+    def __init__(self, dataset='xsub'):
+        self.dataset = dataset
+        if self.dataset not in ['xview', 'xsub', 'kinectics']:
+            raise ValueError(
+                f'The dataset type {self.dataset} is not supported')
+        if self.dataset == 'xview':
+            self.pairs = ((1, 2), (2, 21), (3, 21), (4, 3), (5, 21), (6, 5),
+                          (7, 6), (8, 7), (9, 21), (10, 9), (11, 10), (12, 11),
+                          (13, 1), (14, 13), (15, 14), (16, 15), (17, 1),
+                          (18, 17), (19, 18), (20, 19), (22, 23), (21, 21),
+                          (23, 8), (24, 25), (25, 12))
+        elif self.dataset == 'xsub':
+            self.pairs = ((1, 2), (2, 21), (3, 21), (4, 3), (5, 21), (6, 5),
+                          (7, 6), (8, 7), (9, 21), (10, 9), (11, 10), (12, 11),
+                          (13, 1), (14, 13), (15, 14), (16, 15), (17, 1),
+                          (18, 17), (19, 18), (20, 19), (22, 23), (21, 21),
+                          (23, 8), (24, 25), (25, 12))
+        elif self.dataset == 'kinetics':
+            self.pairs = ((0, 0), (1, 0), (2, 1), (3, 2), (4, 3), (5, 1),
+                          (6, 5), (7, 6), (8, 2), (9, 8), (10, 9), (11, 5),
+                          (12, 11), (13, 12), (14, 0), (15, 0), (16, 14), (17,
+                                                                           15))
+
+    def __call__(self, results):
+        """Performs the Bone formatting.
+
+        Args:
+            results (dict): The resulting dict to be modified and passed
+                to the next transform in pipeline.
+        """
+        keypoint = results['keypoint']
+        M, T, V, C = keypoint.shape
+        bone = np.zeros((M, T, V, C), dtype=np.float32)
+
+        for v1, v2 in self.pairs:
```

Comment:
u serious? Note that for kinetics, there is a pair (0, 0), then it will be (-1, -1)